presto-main/src/main/java/com/facebook/presto/operator/scalar/AbstractArrayMinMaxFunction.java

```diff
@@ -97,7 +96,8 @@ public FunctionInfo specialize(Map<String, Type> types, int arity, TypeManager t
         MethodHandle compareMethodHandle = functionRegistry.resolveOperator(operatorType, ImmutableList.of(elementType, elementType)).getMethodHandle();
         MethodHandle methodHandle = METHOD_HANDLES.get(elementType.getJavaType());
         if (methodHandle == null) {
-            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, String.format("%s is not supported in array_min/max", elementType.getJavaType()));
+            methodHandle = METHOD_HANDLE_OBJECT;
```

Comment:
Do you need to adapt the return type of the method handle to be `elementType.getJavaType()`? I'm not familiar enough with byte code to know if it matters, but I could imagine `array_min(array[array[1]], array[array[2]])[1]` hitting a class cast issue